src/utils/list_cpu_features.c

```diff
@@ -30,7 +36,176 @@
 #include "cpuinfo_ppc.h"
 #endif
 
-static void PrintEscapedAscii(const char* str) {
+// Design principles
+// -----------------
+// We build a tree structure containing all the data to be displayed.
+// Then depending on the output type (text or json) we walk the tree and display
+// the data accordingly.
+
+// We use a bump allocator to allocate strings and nodes of the tree,
+// Memory is not intented to be reclaimed.
+typedef struct {
+  char* ptr;
+  size_t size;
+} BumpAllocator;
+
+// Allocate a buffer of size `size`.
+static BumpAllocator BA_Create(size_t size) {
+  char* const ptr = (char*)malloc(size);
+  BumpAllocator BA;
+  if (ptr) BA = (BumpAllocator){.ptr = ptr, .size = size};
+  return BA;
+}
+
+// Update the available memory left in the BumpAllocator.
+static void* BA_Bump(BumpAllocator* BA, size_t size) {
+  assert(BA->size >= size);
+  void* ptr = BA->ptr;
+  BA->size -= size;
+  BA->ptr += size;
```

Comment:
This allocation doesn't consider alignment requirements, so this is going to return un-aligned addresses that will be cast to pointers to aligned structs (e.g. TreeValueType*), which will crash on certain platforms. A quick fix for this is to ensure the final allocated size is a multiple of 8.